hudi-spark-datasource/hudi-spark-common/src/main/java/org/apache/hudi/commit/BaseDatasetBulkInsertCommitActionExecutor.java

```diff
@@ -56,20 +56,19 @@ public abstract class BaseDatasetBulkInsertCommitActionExecutor implements Seria
 
   protected final transient HoodieWriteConfig writeConfig;
   protected final transient SparkRDDWriteClient writeClient;
-  protected final String instantTime;
+  protected String instantTime;
   protected HoodieTable table;
 
   public BaseDatasetBulkInsertCommitActionExecutor(HoodieWriteConfig config,
-                                                   SparkRDDWriteClient writeClient,
-                                                   String instantTime) {
+                                                   SparkRDDWriteClient writeClient) {
     this.writeConfig = config;
     this.writeClient = writeClient;
-    this.instantTime = instantTime;
   }
 
   protected void preExecute() {
+    instantTime = writeClient.startCommit(getCommitActionType());
+    table = writeClient.initTable(getWriteOperationType(), Option.ofNullable(instantTime));
```

Comment:
Previously `writeClient.initTable` happens before `startCommitWithTime`; the order now (`initTable` after `startCommit`) makes more sense.